sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/XmlSensorTest.java

```diff
@@ -76,6 +77,38 @@
 
   private final RuleKey parsingErrorCheckRuleKey = RuleKey.of(CheckRepository.REPOSITORY_KEY, parsingErrorCheckKey);
 
+  @Test
+  public void testPerformance() throws Exception {
+    init(false, temporaryFolder.getRoot());
+    File smallFile = temporaryFolder.newFile("smallFile.xml");
+    File bigFile = temporaryFolder.newFile("bigFile.xml");
```

Comment:
Extract the logic to build a `xml` file into a dedicated method. So you can configure your call with the number of tags you want. It will simplify the test and allow you to repeat the operation.